### PR TITLE
Export `useResizeObserver` React Native version directly

### PR DESCRIPTION
**Pull request by DaniGuardiola**
## What?

`useResizeObserver` was recently revamped (see https://github.com/WordPress/gutenberg/pull/64943). There were some concerns about the React Native version potentially breaking, this PR addresses that.

## Why?

React Native users could potentially run into a bug if they tried to use the updated version of the utility, since it depends on web-exclusive APIs like `ResizeObserver`.

## How?

Export legacy React Native version at the top-level, as discussed here: https://github.com/WordPress/gutenberg/pull/65389#issuecomment-2360130085

### CODE REVIEW

**Comment by ciampo** [comment]
cc @dcalhoun @geriux

File: packages/compose/src/hooks/use-resize-observer/index.native.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Internal dependencies
+ */
+import useResizeObserver from './legacy/index.native';
+
+export default useResizeObserver;
```
Comment (jsnajdr) [inline packages/compose/src/hooks/use-resize-observer/index.native.js]: Could be just one line: `export { default } from './legacy';`

**Review by jsnajdr — APPROVED** [review]
Thanks for fixing this :+1:

**Comment by ciampo** [comment]
Does this PR need to be backported to the release branch?

File: packages/compose/src/hooks/use-resize-observer/index.native.js (hunk 1 above)
Comment (DaniGuardiola) [reply packages/compose/src/hooks/use-resize-observer/index.native.js]: True, will do that and merge.

**Comment by DaniGuardiola** [comment]
@ciampo 

> Does this PR need to be backported to the release branch?

I'm not sure how this stuff works exactly, but at the very least, there should be no risk of users introducing code that would be made obsolete at a later date/release by this PR's changes. Any users trying to use the wrong (web-exclusive) API in a React Native context will instantly get an error. That said, I miss the context and knowledge to be able to answer that question will full confidence.

**Review by dcalhoun — COMMENTED** [review]
Following up to share that this change looks sound to me. 👍🏻 

There are no errors in the native mobile app without this change currently, as we do not utilize the new callback variant, but it feels worthwhile to guard against this with the merged changes. If someone attempted to use the callback variant, it would result in the following error in the native mobile editor. 

```
 ERROR  ReferenceError: Property 'ResizeObserver' doesn't exist
```

Thank you for merging this change! 🙇🏻

**Pull request merged**